crates/sui-framework/docs/sui_system.md

```diff
@@ -237,8 +237,9 @@ This function will be called only once in genesis.
     max_validator_candidate_count: u64,
     min_validator_stake: u64,
     storage_gas_price: u64,
+    chain_id: u8
```

Comment:
Is a `u8` big enough for this? I thought in https://github.com/MystenLabs/sui/issues/1402, we decided that `TransactionDigest::genesis()` (which would be 32 bytes) was the right candidate for a chain ID, since chains with different genesis digests have distinct object and tx ID's by construction (modulo hash collisions)

Reply:
that's the alternative i was about to comment to use Transaction::genesis(). depends if we want to include it in every transaction for signing, not for transaction data since we can omit, but if we want move struct to be signed, is it efficient to encode it with the digest?

Reply:
I think it's reasonably efficient for non-txn signatures, particularly given that one can always sign `hash(msg | chain_id | domain_separator)` and check that rather than the signature on the raw message.

TBH I am less worried about efficiency than I am multiple sources of truth (e.g., how to map a `u8` chain id to a`Transaction::genesis`, where this mapping lives) and the resulting complexity for devs.